### PR TITLE
add price ticker

### PR DESCRIPTION
**Pull request by yz89**

### CODE REVIEW

**Review by GopherJ — APPROVED** [review]

**Comment by GopherJ** [comment]
so I should subscribe to Orml oracle's storage right?

**Comment by GopherJ** [comment]
it seems that oracle's storage wasn't updated
image

**Comment by yz89** [comment]
> so I should subscribe to Orml oracle's storage right?

Yes, or you can query the RPC in time.

File: pallets/loans/src/lib.rs [hunk 1]
```diff
@@ -446,6 +446,11 @@ pub mod pallet {
                     }
                 }
             });
+
+            // This is used to trigger the price aggregation to update the results to the ORML Oracle Pallet.
+            for currency_id in Currencies::<T>::get().iter() {
+                let _ = Self::get_price(currency_id);
```
Comment (kaichaosun) [inline pallets/loans/src/lib.rs:452]: This needs to be better weighted.

Comment (yz89) [reply pallets/loans/src/lib.rs:452]: Is any suggestion here?

Comment (kaichaosun) [reply pallets/loans/src/lib.rs:452]: The `on_initialized` function should be able to return the weight.

Comment (yz89) [reply pallets/loans/src/lib.rs:452]: We moved the code into `on_finalize` in #219 and maybe needn't anymore.

**Pull request merged**